f/radio/f_radioChannels.sqf

```diff
@@ -0,0 +1,142 @@
+// F3 - Folk ARPS Long-Range Radio Module
+// Credits and documentation: https://github.com/folkarps/F3/wiki
+
+/* ========================
+This module is activated in init.sqf.
+Example:
+[true] execVM "f\radio\f_radioChannels.sqf";
+
+Arguments:
+0. Long-range split mode (boolean). If true, each radio backpack type has a separate channel. If false, all backpacks use one channel. Default false.
+
+This module requires some missionmaker configuration. If you want to enable vehicle radio access, you need to specify vehicle classes or variable names below.
+=========================== */
+
+params [["_splitMode",false]];
+
+// Disable the Command channel to avoid cheating
+2 enableChannel false;
+
+// There is a maximum of 10 channels at any time. Swap in these radios for GM and CSLA missions:
+/*
+	"US85_bpPRC77",
+	"CSLA_bpWpR129",
+	"CSLA_bpWpRF10",
+	"gm_gc_backpack_r105m_brn",
+	"gm_ge_backpack_sem35_oli"
+*/
+
+if (isServer) then {
+
+	// Make the split mode something we can reference later
+	f_var_radioSplitMode = _splitMode;
+	publicVariable "f_var_radioSplitMode";
+
+	// This will be used later
+	f_var_radioChannelUnified = [];
```

Comment:
I'm posting this here to keep it related answers together in the same thread:

I suggest merging the 3 arrays (channel names, backpacks, vehicles) into a singular simple structure:
```sqf
f_var_radioChannels = [
	// Arrays consist of channel name (as shown ingame) and a list of backpacks, vehicle types, or vechile variable names (as strings)
	["LR Channel 1",   ["B_SDV_01_F", "B_RadioBag_01_black_F"]],
	["LR Specialists", ["C_Tractor_01_F", "B_RadioBag_01_mtp_F"]],
];
```

I think it helps if things that belong in the same channel are also specified in the same place

Reply:
At the moment the arrays of eligible classnames are actually Hashmaps of arrays, not just arrays. This is an important distinction because the hashmap key for each channel is also the channel ID, and this is cross-referenced and used in the channel add/remove function. I seriously doubt whether the ID key can be replaced by a customisable channel name in this usage, or whether moving to arrays instead of hashmaps is likely to be an improvement to complexity or performance.

The hashmaps of vehicles and backpacks could be merged easily.

Reply:
I wasn't implying that we should move away from the hashmaps. The above array can be converted (at runtime) to a hashmap easily (if it turns out that channel numbers are different than e.g., `_forEachIndex + 1`).

Reply:
The default configuration should accommodate all template factions, in split mode, which means we need a channel for all 9 vanilla backpacks, which is near to 10 as makes no odds. The alternative is to go back to having duplicate generation methods based on which mode we're running in, which I'm not a fan of.

Reply:
I agree that the default should accommodate everything. But right now it is hard to use custom channels (for other purposes) when this module is active. The added code complexity is worth it to me for the increased compatibility and ease of use

Reply:
This module currently assumes that no other code (which may run at the same time) calls radioChannelCreate because we don't properly check/use the return values.

Reply:
I think I found a semi-elegant solution

Reply:
I'm not sure it has to be a parameter since we can just take the information from the `f_var_radioChannels`.

I'm not a fan of having the channel IDs in the same structure that the missionmaker modifies. If a missionmaker comments out one of these lines `f_var_radioChannels set [...];` then the channel numbers may be mismatched compared to the ones that we actually create. 
If we keep it like this, we want to have checks in place to prevent these issues. That is, IDs returned from `radioChannelCreate` must match the keys in `f_var_radioChannels`. This should be as simple as comparing `_ChannelID` with `_i + some_offset`.

Reply:
We could also print some warning/error if the provided number (`_channelCount`) is smaller than the non-empty entries in `f_var_radioChannels `.

Reply:
A missionmaker commenting out random lines from the middle of an F3 script can break pretty much anything. There is no ultimate safety against code destruction.
If you want to put the brainpower into figuring out how to make that work then go for it, but I don't.